test/sql/json/read_ndjson.test

```diff
@@ -14,10 +29,70 @@ SELECT * FROM read_json_objects('data/json/example.ndjson')
 {"id":5,"name":"Raising Arizona"}
 
 query I
-SELECT * FROM read_ndjson_objects('data/json/example.ndjson')
+SELECT * FROM read_ndjson_objects('data/json/example_n.ndjson')
 ----
 {"id":1,"name":"O Brother, Where Art Thou?"}
 {"id":2,"name":"Home for the Holidays"}
 {"id":3,"name":"The Firm"}
 {"id":4,"name":"Broadcast News"}
 {"id":5,"name":"Raising Arizona"}
+
+# example_r is with \r newlines
+query I
+SELECT * FROM read_json_objects('data/json/example_r.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+query I
+SELECT * FROM read_ndjson_objects('data/json/example_r.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+# example_rn is with \r\n newlines
+query I
+SELECT * FROM read_json_objects('data/json/example_rn.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+query I
+SELECT * FROM read_ndjson_objects('data/json/example_rn.ndjson')
```

Comment:
Could we add tests where we:

* Read json from a gzip compressed file (`example_rn.ndjson.gz`)
* Read json over a network connection using HTTPFS (you can host files [here](https://github.com/duckdb/duckdb-data/releases/tag/v1.0) for that purpose)
* Read json from stdin in the shell (see `tools/shell/shell-test.py:894` where we read CSVs from `/dev/stdin`)

Malformed tests, including checking that the expected error is thrown (using the new `statement error` syntax):

* Read an empty file
* Unterminated quotes
* Duplicate JSON key
* Multiple JSON objects on one line
* What if we try to read a CSV file using the JSON functions?
* What if we try to read a Parquet file using the JSON functions?

Finally, perhaps out of scope for this PR, but for testing it is extremely useful to also have a writer because you can do round-trip verification and generate large datasets without having to check them in. This is used quite extensively in the CSV and Parquet readers/writers. For example, you could write the TPC-H database as JSON files, then read them back in and verify that the query results are still correct.

Reply:
I've implemented pretty much all of this, and it required parsing the JSON before ingesting it (which is technically not required for NDJSON), so this is a slight performance hit.

Much less than I expected, though, it now takes ~1.21s instead of ~0.73 seconds :) And we can now throw descriptive errors.

Reply:
Also, we should be able to create a `COPY ... (FORMAT JSON)` really easily - all the components are already there:
```sql
copy (select to_json(lineitem) from lineitem) to '__TEST_DIR__/lineitem.json' (format csv, quote '')
```

Maybe we can do some kind of rewrite when `(FORMAT JSON)` is specified?